core/include/vecmem/memory/device_atomic_ref.hpp

```diff
@@ -40,7 +51,7 @@ namespace vecmem {
 using memory_order = std::memory_order;
 
 /// @c vecmem::atomic_ref equals @c std::atomic_ref in host code with C++20
-template <typename T>
+template <typename T, address_space address = address_space::global>
```

Comment:
You should be able to avoid warnings about an unused template parameter by not giving the value a name. Like:

```c++
template <typename T, device_address_space = device_address_space::global>
```

(Just like you would for regular function arguments.)

Though I'm not 1000% sure about this... :confused: